### DISCUSSION TITLE
DFM17 USB Connector?

### DISCUSSION THREAD

**Discussion by K4KDR** [mikaelnousiainen/RS41ng#67] [General]
Good day and many thanks for continuing development of the **dfm17 branch**!

I have cloned that branch separately and it compiles fine.  Today I'll follow your much appreciated instructions on how to connect to the proper programming pins.

But out of curiosity, is the USB connector on the DFM-17 no use at all to us?    It would be SO convenient if that was also a programming and/or testing interface that we could utilize.

Thanks!

-Scott,  K4KDR

**Comment by mikaelnousiainen** [comment]
Thanks! I will be merging the dfm17 branch soon to the main branch -- after some basic testing on both DFM-17 and RS41.

The USB connector cannot be used for programming. It is actually a serial port (with a very non-standard connector) as described here: https://wiki.recessim.com/view/DFM-17_Radiosonde#Mini_USB_Port 

The serial port could of course be used for similar purposes as on the RS41, e.g. to output GPS NMEA data.

**Reply by Dr-HH** [reply]
Using the mini USB on the DFM-17 I am able to read on PowerShell the board "Version GRAWMET 5.16.04.22"
How can access the program to modify the frequency settings through this port?

**Comment by K4KDR** [comment]
Many thanks!

**Comment by K4KDR** [comment]
Success!    Still need to test near a window to confirm that the GPS is working properly, but 4FSK -and- APRS both working on a DFM-17.

Thanks so much for adding support for this platform!

![DFM-17_4FSK_test](https://github.com/mikaelnousiainen/RS41ng/assets/19886368/1c676bc9-84f3-41c7-94a6-800d5d429eb4)
.
![DFM-17_APRS_test](https://github.com/mikaelnousiainen/RS41ng/assets/19886368/5e719b46-0b26-4f4c-8349-e96a60d33bb8)

**Comment by kd2eat** [comment]
The USB port CAN be used to provide power, but not programming.
